### PR TITLE
docs: Mention new config system is complete

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

1. Updated the note on the new config file page to indicate that it's feature complete.
2. Added a note on the old config file page to indicate that we are transitioning to a new config system.

#### Is there anything you'd like reviewers to focus on?

Is the wording clear enough?

### CODE REVIEW

File: docs/src/use/configure/configuration-files.md
Comment (amareshsm) [inline docs/src/use/configure/configuration-files.md:12]:
```suggestion
We are transitioning to a new config system in ESLint v9.0.0. The config system shared on this page is currently the default but will be deprecated in v9.0.0. You can opt-in to the new config system by following the instructions in the [documentation](configuration-files-new).
```

**Review by amareshsm — APPROVED** [review]
The wording looks good to me. Waiting for the code owner's review.

**Review by mdjermanovic — APPROVED** [review]

**Comment by snitin315** [comment]
Resubmitted in https://github.com/eslint/eslint/pull/17267

**Pull request merged**
